### PR TITLE
Waypoint State Unit Test

### LINKED ISSUE #142: Waypoint state unit test
# Waypoint Unit Test

## Problem

Once the waypoint state is implemented, we want a file to test the functionality of said state.

## Solution

Make a unit test that takes an input of a set of waypoints in the format of the SUAS waypoints and fly to each waypoint using the waypoint state

### PR DESCRIPTION
**Pull request by RexBerry**
Waypoint state unit test and fixes to the waypoint and takeoff states

### CODE REVIEW

File: flight/test_files/waypoint_pathfinding_test.py [hunk 1]
```diff
@@ -214,6 +269,10 @@ def main() -> None:
         )
         boundary_vertices.append(Point(easting, northing))
 
+    # Generate random waypoints to use for the waypoint state unit test
+    # Prints json data to be pasted in ../data/waypoint_data.json
+    print(generate_random_waypoints(boundary_vertices, force_zone_number, force_zone_letter))
```
Comment (cjhr95) [inline flight/test_files/waypoint_pathfinding_test.py]: Do we really want to print all 100 randomly generated waypoints?

File: flight/test_files/waypoint_pathfinding_test.py [hunk 2]
```diff
@@ -187,6 +195,53 @@ def to_svg_coord(x: float, y: float) -> tuple[float, float]:
...
+        running the waypoint state unit test.
+    """
+    waypoints: list[CoordinateWithAltitude] = []
+    for _ in range(100):
+        random_point: Point = random_point_in_shape(vertices)
+
+        latitude: float
+        longitude: float
+        latitude, longitude = utm.to_latlon(
+            random_point.x, random_point.y, utm_zone_number, utm_zone_letter
```
Comment (cjhr95) [inline flight/test_files/waypoint_pathfinding_test.py:235]: Do you have a way to generate points for the golf course?

**Review by cjhr95 — CHANGES_REQUESTED** [review]

File: flight/test_files/waypoint_pathfinding_test.py (hunk 1 above)
Comment (RexBerry) [reply flight/test_files/waypoint_pathfinding_test.py]: I can change it if necessary. This was just a quick way to generate random waypoints for the waypoint state unit test. I would run the waypoint pathfinding test (which only tests pathfinding) and then copy the printed waypoints into waypoint_data.json.

File: flight/test_files/waypoint_pathfinding_test.py (hunk 2 above)
Comment (RexBerry) [reply flight/test_files/waypoint_pathfinding_test.py:235]: No

**Comment by RexBerry** [comment]
I will add some things Tuesday. I also noticed a bug in the boundary checking function in the unit test.

**Comment by cjhr95** [comment]
Also, something from the flight day, add logging for which waypoint you're going to, like a number or the coordinates. It'll help with the path visualization on QGroundControl.

**Comment by RexBerry** [comment]
I will rewrite the test code that's supposed to check whether the drone got close enough to the waypoint.

**Review by Ooglely — DISMISSED** [review]
Looks good to me as of now, but I know you are making changes

File: flight/test_files/waypoint_pathfinding_test.py (hunk 2 above)
Comment (RexBerry) [reply flight/test_files/waypoint_pathfinding_test.py:235]: In line 257 of flight/test_files/waypoint_pathfinding_test, `waypoint_data.json` can be replaced with `golf_data.json`. Running `python3 -m flight.test_files.waypoint_pathfinding_test` will then print 100 random waypoints within the golf course that can then be pasted into flight/data/golf_data.json.

**Review by cjhr95 — APPROVED** [review]

**Pull request merged**